backend/typescript/services/implementations/reviewService.ts

```diff
@@ -32,6 +37,156 @@ class ReviewService implements IReviewService {
     if (db !== sequelize) sequelize.close(); // Using test db instead of main db
   }
 
+  /* eslint-disable class-methods-use-this, no-await-in-loop */
+  async findOrCreateTag(
+    review: PgReview,
+    tag: TagRequest,
+    t: Transaction,
+  ): Promise<PgTag> {
+    const newTag = await PgTag.findOrCreate({
+      where: { name: tag.name },
+      transaction: t,
+    }).then((data) => data[0]);
+    await review.$add("tags", newTag, { transaction: t });
+    return newTag;
+  }
+
+  async findOrCreateTags(
+    review: PgReview,
+    tags: TagRequest[],
+    t: Transaction,
+  ): Promise<TagResponse[]> {
+    const tagsRet: TagResponse[] = [];
```

Comment:
does it make sense for this to be a `const`? we're adding to the list so the list is changing

Reply:
The const refers to the reference to the array which is a constant. Adding elements to the array doesn't change the reference.